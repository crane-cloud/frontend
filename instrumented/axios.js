function cov_12zl524xrh() {
  var path = "/home/greatest/Desktop/code/cranecloud/frontend/em/frontend/src/axios.js";
  var hash = "4ba0e69af2be53c2e422da5c94744e69b996097f";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "/home/greatest/Desktop/code/cranecloud/frontend/em/frontend/src/axios.js",
    statementMap: {
      "0": {
        start: {
          line: 4,
          column: 17
        },
        end: {
          line: 6,
          column: 2
        }
      },
      "1": {
        start: {
          line: 8,
          column: 0
        },
        end: {
          line: 10,
          column: 4
        }
      },
      "2": {
        start: {
          line: 12,
          column: 0
        },
        end: {
          line: 26,
          column: 2
        }
      },
      "3": {
        start: {
          line: 13,
          column: 16
        },
        end: {
          line: 13,
          column: 24
        }
      },
      "4": {
        start: {
          line: 15,
          column: 12
        },
        end: {
          line: 15,
          column: 17
        }
      },
      "5": {
        start: {
          line: 17,
          column: 4
        },
        end: {
          line: 23,
          column: 5
        }
      },
      "6": {
        start: {
          line: 18,
          column: 6
        },
        end: {
          line: 18,
          column: 27
        }
      },
      "7": {
        start: {
          line: 19,
          column: 6
        },
        end: {
          line: 19,
          column: 33
        }
      },
      "8": {
        start: {
          line: 20,
          column: 11
        },
        end: {
          line: 23,
          column: 5
        }
      },
      "9": {
        start: {
          line: 21,
          column: 6
        },
        end: {
          line: 21,
          column: 56
        }
      },
      "10": {
        start: {
          line: 22,
          column: 6
        },
        end: {
          line: 22,
          column: 41
        }
      },
      "11": {
        start: {
          line: 24,
          column: 4
        },
        end: {
          line: 24,
          column: 29
        }
      }
    },
    fnMap: {
      "0": {
        name: "(anonymous_0)",
        decl: {
          start: {
            line: 13,
            column: 2
          },
          end: {
            line: 13,
            column: 3
          }
        },
        loc: {
          start: {
            line: 13,
            column: 16
          },
          end: {
            line: 13,
            column: 24
          }
        },
        line: 13
      },
      "1": {
        name: "(anonymous_1)",
        decl: {
          start: {
            line: 14,
            column: 2
          },
          end: {
            line: 14,
            column: 3
          }
        },
        loc: {
          start: {
            line: 14,
            column: 13
          },
          end: {
            line: 25,
            column: 3
          }
        },
        line: 14
      }
    },
    branchMap: {
      "0": {
        loc: {
          start: {
            line: 17,
            column: 4
          },
          end: {
            line: 23,
            column: 5
          }
        },
        type: "if",
        locations: [{
          start: {
            line: 17,
            column: 4
          },
          end: {
            line: 23,
            column: 5
          }
        }, {
          start: {
            line: 17,
            column: 4
          },
          end: {
            line: 23,
            column: 5
          }
        }],
        line: 17
      },
      "1": {
        loc: {
          start: {
            line: 17,
            column: 8
          },
          end: {
            line: 17,
            column: 70
          }
        },
        type: "binary-expr",
        locations: [{
          start: {
            line: 17,
            column: 8
          },
          end: {
            line: 17,
            column: 37
          }
        }, {
          start: {
            line: 17,
            column: 41
          },
          end: {
            line: 17,
            column: 70
          }
        }],
        line: 17
      },
      "2": {
        loc: {
          start: {
            line: 20,
            column: 11
          },
          end: {
            line: 23,
            column: 5
          }
        },
        type: "if",
        locations: [{
          start: {
            line: 20,
            column: 11
          },
          end: {
            line: 23,
            column: 5
          }
        }, {
          start: {
            line: 20,
            column: 11
          },
          end: {
            line: 23,
            column: 5
          }
        }],
        line: 20
      }
    },
    s: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0,
      "5": 0,
      "6": 0,
      "7": 0,
      "8": 0,
      "9": 0,
      "10": 0,
      "11": 0
    },
    f: {
      "0": 0,
      "1": 0
    },
    b: {
      "0": [0, 0],
      "1": [0, 0],
      "2": [0, 0]
    },
    _coverageSchema: "1a1c01bbd47fc00a2c39e90264f33305004495a9",
    hash: "4ba0e69af2be53c2e422da5c94744e69b996097f"
  };
  var coverage = global[gcv] || (global[gcv] = {});
  if (!coverage[path] || coverage[path].hash !== hash) {
    coverage[path] = coverageData;
  }
  var actualCoverage = coverage[path];
  {
    // @ts-ignore
    cov_12zl524xrh = function () {
      return actualCoverage;
    };
  }
  return actualCoverage;
}
cov_12zl524xrh();
import axios from "axios";
import { API_BASE_URL } from "./config";
const instance = (cov_12zl524xrh().s[0]++, axios.create({
  baseURL: API_BASE_URL
}));
cov_12zl524xrh().s[1]++;
instance.defaults.headers.Authorization = `Bearer ${localStorage.getItem("token")}`;
cov_12zl524xrh().s[2]++;
instance.interceptors.response.use(response => {
  cov_12zl524xrh().f[0]++;
  cov_12zl524xrh().s[3]++;
  return response;
}, error => {
  cov_12zl524xrh().f[1]++;
  let e = (cov_12zl524xrh().s[4]++, error);
  // this checks for when a token is not verified and logs you out
  cov_12zl524xrh().s[5]++;
  if ((cov_12zl524xrh().b[1][0]++, error.response.status === 401) || (cov_12zl524xrh().b[1][1]++, error.response.status === 422)) {
    cov_12zl524xrh().b[0][0]++;
    cov_12zl524xrh().s[6]++;
    localStorage.clear();
    cov_12zl524xrh().s[7]++;
    window.location.href = "/";
  } else {
    cov_12zl524xrh().b[0][1]++;
    cov_12zl524xrh().s[8]++;
    if (error.response.status === 502) {
      cov_12zl524xrh().b[2][0]++;
      cov_12zl524xrh().s[9]++;
      e = "Your request too long possible Server Error.";
      cov_12zl524xrh().s[10]++;
      window.location.href = "/projects";
    } else {
      cov_12zl524xrh().b[2][1]++;
    }
  }
  cov_12zl524xrh().s[11]++;
  return Promise.reject(e);
});
export default instance;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJjb3ZfMTJ6bDUyNHhyaCIsImFjdHVhbENvdmVyYWdlIiwiYXhpb3MiLCJBUElfQkFTRV9VUkwiLCJpbnN0YW5jZSIsInMiLCJjcmVhdGUiLCJiYXNlVVJMIiwiZGVmYXVsdHMiLCJoZWFkZXJzIiwiQXV0aG9yaXphdGlvbiIsImxvY2FsU3RvcmFnZSIsImdldEl0ZW0iLCJpbnRlcmNlcHRvcnMiLCJyZXNwb25zZSIsInVzZSIsImVycm9yIiwiZiIsImUiLCJiIiwic3RhdHVzIiwiY2xlYXIiLCJ3aW5kb3ciLCJsb2NhdGlvbiIsImhyZWYiLCJQcm9taXNlIiwicmVqZWN0Il0sInNvdXJjZXMiOlsiYXhpb3MuanMiXSwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGF4aW9zIGZyb20gXCJheGlvc1wiO1xuaW1wb3J0IHsgQVBJX0JBU0VfVVJMIH0gZnJvbSBcIi4vY29uZmlnXCI7XG5cbmNvbnN0IGluc3RhbmNlID0gYXhpb3MuY3JlYXRlKHtcbiAgYmFzZVVSTDogQVBJX0JBU0VfVVJMLFxufSk7XG5cbmluc3RhbmNlLmRlZmF1bHRzLmhlYWRlcnMuQXV0aG9yaXphdGlvbiA9IGBCZWFyZXIgJHtsb2NhbFN0b3JhZ2UuZ2V0SXRlbShcbiAgXCJ0b2tlblwiXG4pfWA7XG5cbmluc3RhbmNlLmludGVyY2VwdG9ycy5yZXNwb25zZS51c2UoXG4gIChyZXNwb25zZSkgPT4gcmVzcG9uc2UsXG4gIChlcnJvcikgPT4ge1xuICAgIGxldCBlID0gZXJyb3I7XG4gICAgLy8gdGhpcyBjaGVja3MgZm9yIHdoZW4gYSB0b2tlbiBpcyBub3QgdmVyaWZpZWQgYW5kIGxvZ3MgeW91IG91dFxuICAgIGlmIChlcnJvci5yZXNwb25zZS5zdGF0dXMgPT09IDQwMSB8fCBlcnJvci5yZXNwb25zZS5zdGF0dXMgPT09IDQyMikge1xuICAgICAgbG9jYWxTdG9yYWdlLmNsZWFyKCk7XG4gICAgICB3aW5kb3cubG9jYXRpb24uaHJlZiA9IFwiL1wiO1xuICAgIH0gZWxzZSBpZihlcnJvci5yZXNwb25zZS5zdGF0dXMgPT09IDUwMil7XG4gICAgICBlID0gXCJZb3VyIHJlcXVlc3QgdG9vIGxvbmcgcG9zc2libGUgU2VydmVyIEVycm9yLlwiXG4gICAgICB3aW5kb3cubG9jYXRpb24uaHJlZiA9IFwiL3Byb2plY3RzXCI7XG4gICAgfVxuICAgIHJldHVybiBQcm9taXNlLnJlamVjdChlKTtcbiAgfVxuKTtcblxuZXhwb3J0IGRlZmF1bHQgaW5zdGFuY2U7XG4iXSwibWFwcGluZ3MiOiI7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7OztJQWVZO0lBQUFBLGNBQUEsWUFBQUEsQ0FBQTtNQUFBLE9BQUFDLGNBQUE7SUFBQTtFQUFBO0VBQUEsT0FBQUEsY0FBQTtBQUFBO0FBQUFELGNBQUE7QUFmWixPQUFPRSxLQUFLLE1BQU0sT0FBTztBQUN6QixTQUFTQyxZQUFZLFFBQVEsVUFBVTtBQUV2QyxNQUFNQyxRQUFRLElBQUFKLGNBQUEsR0FBQUssQ0FBQSxPQUFHSCxLQUFLLENBQUNJLE1BQU0sQ0FBQztFQUM1QkMsT0FBTyxFQUFFSjtBQUNYLENBQUMsQ0FBQztBQUFDSCxjQUFBLEdBQUFLLENBQUE7QUFFSEQsUUFBUSxDQUFDSSxRQUFRLENBQUNDLE9BQU8sQ0FBQ0MsYUFBYSxHQUFJLFVBQVNDLFlBQVksQ0FBQ0MsT0FBTyxDQUN0RSxPQUFPLENBQ1AsRUFBQztBQUFDWixjQUFBLEdBQUFLLENBQUE7QUFFSkQsUUFBUSxDQUFDUyxZQUFZLENBQUNDLFFBQVEsQ0FBQ0MsR0FBRyxDQUMvQkQsUUFBUSxJQUFLQTtFQUFBQSx1QkFBQTtFQUFBQSx1QkFBQTtFQUFBQSxNQUFBLENBQUFBLFFBQVE7QUFBRCxDQUFDLEVBQ3JCRSxLQUFLLElBQUs7RUFBQWhCLGNBQUEsR0FBQWlCLENBQUE7RUFDVCxJQUFJQyxDQUFDLElBQUFsQixjQUFBLEdBQUFLLENBQUEsT0FBR1csS0FBSztFQUNiO0VBQUFoQixjQUFBLEdBQUFLLENBQUE7RUFDQSxJQUFJLENBQUFMLGNBQUEsR0FBQW1CLENBQUEsVUFBQUgsS0FBSyxDQUFDRixRQUFRLENBQUNNLE1BQU0sS0FBSyxHQUFHLE1BQUFwQixjQUFBLEdBQUFtQixDQUFBLFVBQUlILEtBQUssQ0FBQ0YsUUFBUSxDQUFDTSxNQUFNLEtBQUssR0FBRyxHQUFFO0lBQUFwQixjQUFBLEdBQUFtQixDQUFBO0lBQUFuQixjQUFBLEdBQUFLLENBQUE7SUFDbEVNLFlBQVksQ0FBQ1UsS0FBSyxFQUFFO0lBQUNyQixjQUFBLEdBQUFLLENBQUE7SUFDckJpQixNQUFNLENBQUNDLFFBQVEsQ0FBQ0MsSUFBSSxHQUFHLEdBQUc7RUFDNUIsQ0FBQyxNQUFNO0lBQUF4QixjQUFBLEdBQUFtQixDQUFBO0lBQUFuQixjQUFBLEdBQUFLLENBQUE7SUFBQSxJQUFHVyxLQUFLLENBQUNGLFFBQVEsQ0FBQ00sTUFBTSxLQUFLLEdBQUcsRUFBQztNQUFBcEIsY0FBQSxHQUFBbUIsQ0FBQTtNQUFBbkIsY0FBQSxHQUFBSyxDQUFBO01BQ3RDYSxDQUFDLEdBQUcsOENBQThDO01BQUFsQixjQUFBLEdBQUFLLENBQUE7TUFDbERpQixNQUFNLENBQUNDLFFBQVEsQ0FBQ0MsSUFBSSxHQUFHLFdBQVc7SUFDcEMsQ0FBQztNQUFBeEIsY0FBQSxHQUFBbUIsQ0FBQTtJQUFBO0VBQUQ7RUFBQ25CLGNBQUEsR0FBQUssQ0FBQTtFQUNELE9BQU9vQixPQUFPLENBQUNDLE1BQU0sQ0FBQ1IsQ0FBQyxDQUFDO0FBQzFCLENBQUMsQ0FDRjtBQUVELGVBQWVkLFFBQVEifQ==