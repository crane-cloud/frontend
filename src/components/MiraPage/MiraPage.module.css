.FormHeading {
  padding: 0.3rem 0rem;
  font-weight: 600;
}

.FormFieldWithTooltip {
  /* width: 60%; same as input field */
  display: flex;
  position: relative;
}

.FormInputTooltipContainer {
  position: absolute;
  right: 0%;
  padding-right: 1rem;
  padding-top: 1rem;
  display: flex;
  align-items: center;
}

.HeadingWithTooltip {
  display: flex;
}

.HeadingWithTooltip h4 {
  margin: 0;
  padding: 0.5rem 0;
}

.Dropzone {
  /* width: 570px; */
  color: #808080;
  padding: 1rem 2rem;
  height: 16rem;
  background-color: whitesmoke;
  border: 3px dashed #aeaeae;
  cursor: pointer;
}

.DropSection {
  padding-bottom: 2rem;
}

.FormInputs {
  height: calc(100vh - 12rem);
}

.FormInputs::-webkit-scrollbar {
  width: 4px;
  height: 0px;
}

.FormInputs::-webkit-scrollbar-track {
  background: #f1f1f1;
}

.FormInputs::-webkit-scrollbar-thumb {
  background: #555;
}

.FormInputs::-webkit-scrollbar-thumb:hover {
  background: #000;
}

.FrameworkSelect,
.RegistrySelect,
.FormFieldWithTooltip {
  padding-bottom: 1.5rem;
}

.ButtonSection {
  padding-bottom: 5rem;
}
