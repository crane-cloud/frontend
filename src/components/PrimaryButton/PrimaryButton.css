.Primary-Btn {
  background-color: #f7b21f;
  width: 9.7rem;
  height: 2.9rem;
  color: white;
  font-size: 1rem;
  font-weight: bold;
  line-height: 19px;
  text-align: center;
  border: none;
  outline: 1px solid #f7b21f;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.Primary-Btn:hover {
  background-color: transparent;
  color: #f7b21f;
  outline: 1px solid #f7b21f;
}

.Primary-Btn.CancelBtn {
  outline: 1px solid #000000;
  color: #000000;
  background-color: #ffffff;
}

.Primary-Btn.CancelBtn:hover {
  color: #fff;
  background-color: #000;
}
