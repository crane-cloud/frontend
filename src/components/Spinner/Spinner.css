/*! SMALL SPINNER */
.SmallSpinnerWrapper {
  display: flex;
  justify-content: center;
  align-items: center;
}

.Spinner {
  border: 2px solid white;
  width: 1rem;
  height: 1rem;
  border-radius: 100%;
  border-top: 2px solid var(--secondary-color);
  animation: spin 600ms linear infinite;
}

.SpinnerBig {
  width: 2rem;
  height: 2rem;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
