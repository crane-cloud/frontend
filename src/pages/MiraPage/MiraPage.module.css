.FormHeading {
  padding: 0.3rem 0rem;
  font-weight: 600;
}

.FormFieldWithTooltip {
  /* width: 60%; same as input field */
  display: flex;
  position: relative;
}

.FormInputTooltipContainer {
  position: absolute;
  right: 0%;
  padding-right: 1rem;
  padding-top: 1rem;
  display: flex;
  align-items: center;
}

.HeadingWithTooltip {
  display: flex;
}

.HeadingWithTooltip h4 {
  margin: 0;
  padding: 0.5rem 0;
}

.Dropzone {
  /* width: 570px; */
  color: #808080;
  padding: 1rem 2rem;
  height: 16rem;
  background-color: whitesmoke;
  border: 3px dashed #aeaeae;
  cursor: pointer;
}

.DropSection {
  padding-bottom: 2rem;
}
.CreateFormHolder {
  height: calc(100vh - 14.5rem);
}

.FormInputs {
  background: #ffffff;
  padding: 0.7rem;
  border-radius: 0.7rem;
}

.FormInputs::-webkit-scrollbar {
  width: 4px;
  height: 0px;
}

.FormInputs::-webkit-scrollbar-track {
  background: var(--gray-bg-light-color);
}

.FormInputs::-webkit-scrollbar-thumb {
  background: #555;
}

.FormInputs::-webkit-scrollbar-thumb:hover {
  background: #000;
}

.FrameworkSelect,
.RegistrySelect,
.FormFieldWithTooltip {
  padding-bottom: 1.5rem;
}

.ButtonSection {
  padding-bottom: 1rem;
}
