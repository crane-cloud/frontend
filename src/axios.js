import axios from "axios";
import { API_BASE_URL, ACTIVITY_LOGS_API_URL, MONITORING_API_URL } from "./config";



// Create main backend axios instance
const instance = axios.create({
  baseURL: API_BASE_URL,
});
// Create activity axios instance
const userActivityLoggerAxios = axios.create({
  baseURL: ACTIVITY_LOGS_API_URL,
});

// Create monitoring axios instance
const monitoringAxios = axios.create({
  baseURL: MONITORING_API_URL,
});

// Set default headers
const token = localStorage.getItem("token");
instance.defaults.headers.Authorization = `Bearer ${token}`;
userActivityLoggerAxios.defaults.headers.Authorization = `Bearer ${token}`;
monitoringAxios.defaults.headers.Authorization = `Bearer ${token}`;

// Define response interceptor function
const responseInterceptor = (response) => response;

const errorInterceptor = (error) => {
  let e = error;
  // this checks for when a token is not verified and logs you out
  if (error.response.status === 401 || error.response.status === 422) {
    localStorage.clear();
    window.location.href = "/";
  } else if (error.response.status === 502) {
    e = "Your request took too long, possibly a Server Error.";
    window.location.href = "/projects";
  }
  return Promise.reject(e);
};

// Add response interceptors to both instances
instance.interceptors.response.use(responseInterceptor, errorInterceptor);
userActivityLoggerAxios.interceptors.response.use(
  responseInterceptor,
  errorInterceptor
);
monitoringAxios.interceptors.response.use(
  responseInterceptor,
  errorInterceptor
);

export { instance, userActivityLoggerAxios, monitoringAxios };
export default instance;
