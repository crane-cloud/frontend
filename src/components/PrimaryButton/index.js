import React from "react";
import "./PrimaryButton.css";
import PropTypes from "prop-types";

const PrimaryButton = (props) => {
  const { children, className, btntype, color, small, transparent, noPadding } = props;

  const getColorClass = () => {
    switch (color) {
      case "primary":
        return "PrimaryColorBtn";
      case "primary-outline":
        return "PrimaryOutlineColorBtn";
      case "black":
        return "BlackPrimaryBtn";
      case "red":
        return "PrimaryDeleteBtn";
      case "red-outline":
        return "PrimaryDeleteBtnOutline";
      default:
        return "primary-button--blue";
    }
  };
  return (
    <button
      {...props}
      className={`Primary-Btn ${className} ${btntype === "close" && "DeleteBtnOutline"}
        ${color && getColorClass()} ${small===true ? "SmallBtn" : ""} ${transparent && "TransparentBtn"}
        ${noPadding && "NoPaddingBtn"}
      `}
    >
      {children}
    </button>
  );
};
PrimaryButton.propTypes = {
  small: PropTypes.bool,
};
PrimaryButton.defaultProps = {
  small: false,
};

export default PrimaryButton;
