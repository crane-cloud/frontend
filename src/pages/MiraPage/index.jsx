import React, { useState } from "react";
import axios from "axios";
import Dropzone from "../../components/DropZone";
import styles from "./MiraPage.module.css";
import BlackInputText from "../../components/BlackInputText";
import Select from "../../components/Select";
import PrimaryButton from "../../components/PrimaryButton";
import Tooltip from "../../components/Tooltip";
import { MIRA_API_URL } from "../../config";
import Spinner from "../../components/Spinner";
import { retrieveFrameworkChoices } from "../../helpers/frameworkChoices";
import { retrieveRegistryChoices } from "../../helpers/registryChoices";


const MiraPge = ({ projectID }) => {
  const frameworks = retrieveFrameworkChoices();
  const registries = retrieveRegistryChoices();

  const [files, setFiles] = useState([]);
  const [framework, setFramework] = useState("");
  const [registry, setRegistry] = useState("");
  const [image, setImage] = useState({
    name: "",
    version: "",
  });
  const [loading, setLoader] = useState(false);
  const [error, setError] = useState("");

  const getPathName = (path) => {
    let filePath = path.replaceAll("/", "|").replace("|", "");
    filePath = filePath.substring(filePath.indexOf("|")).replace("|", "");
    return filePath;
  };
  const handleChange = ({ target }) => {
    const { name, value } = target;
    if (error) {
      setError("");
    }
    setImage({
      ...image,
      [name]: value,
    });
  };
  const handleDropdownChange = (selected) => {
    setFramework(selected.value);
  };

  const handleRegistryDropdownChange = (selected) => {
    setRegistry(selected.value);
  };

  const handleSubmit = () => {
    setLoader(true);
    const token = localStorage.getItem("token");
    const formData = new FormData();
    const { name, version } = image;
    for (const file of files) {
      const { path } = file;
      formData.append("files", file, getPathName(path));
    }
    formData.append("name", name);
    formData.append("tag", version);
    formData.append("framework", framework);
    formData.append("token", token);
    formData.append("project", projectID);
    formData.append("registry", registry);

    axios
      .post(`${MIRA_API_URL}/containerize`, formData)
      .then(function ({ res }) {
        setLoader(false);
        window.location.href = `/projects/${projectID}/Apps`;
      })
      .catch(function (error) {
        setLoader(false);
        setError("failed to deploy");
      });
  };

  return (
    <div className={styles.CreateFormHolder}>
      <div className={styles.FormInputs}>
        <div className={styles.FormHeading}>Fields marked * are required</div>
        <div className={styles.FrameworkSelect}>
          <Select
            placeholder="Choose a framework"
            options={frameworks}
            onChange={handleDropdownChange}
          />
        </div>
        {framework === "Django" && (
          <div>
            If deploying a Django app also see these additional pre-deployment
            <a href="https://docs.google.com/document/d/1-zqaLC4x4yZflRS-LMhycVbhpCEvyId0smaqAwC5TBE/edit?usp=sharing">
              instructions
            </a>{" "}
          </div>
        )}
        {framework === "Laravel-custom" && (
          <div>
            Please make sure your project has a custom dockerfile added in the
            root of your Laravel app<br></br>
            <a href="https://medium.com/cranecloud/dockerizing-a-laravel-application-36b5ccd23691">
              Take an example
            </a>{" "}
            <br></br> Be sure to use your current version of laravel in your
            dockerfile{" "}
          </div>
        )}
        <div className={styles.RegistrySelect}>
          <Select
            placeholder="Select a registry"
            options={registries}
            onChange={handleRegistryDropdownChange}
          />
        </div>
        <div className={styles.FormFieldWithTooltip}>
          <BlackInputText
            required
            name="name"
            placeholder="Image Name"
            onChange={handleChange}
            value={image.name}
          />
          <div className={styles.FormInputTooltipContainer}>
            <Tooltip
              showIcon
              message="This is the image repository for your image"
              position="left"
            />
          </div>
        </div>
        <div className={styles.FormFieldWithTooltip}>
          <BlackInputText
            placeholder="Version"
            name="version"
            onChange={handleChange}
            value={image.version}
          />
          <div className={styles.FormInputTooltipContainer}>
            <Tooltip
              showIcon
              message="This is preferably a tag for your image"
              position="left"
            />
          </div>
        </div>

        <div className={styles.HeadingWithTooltip}>
          <h4>Upload Zip file</h4>
          <Tooltip
            showIcon
            message="This is the zipped folder containing your source code"
          />
        </div>
        <div className={styles.DropSection}>
          <div className={styles.Dropzone}>
            <Dropzone handleDrop={(files) => setFiles(files)} />
          </div>
        </div>
        {error && <div className="LoginErrorDiv">{error}</div>}
        <div className={styles.ButtonSection}>
          <PrimaryButton className="AuthBtn FullWidth" onClick={handleSubmit}>
            {loading ? <Spinner /> : "Deploy"}
          </PrimaryButton>
        </div>
      </div>
    </div>
  );
};

export default MiraPge;
