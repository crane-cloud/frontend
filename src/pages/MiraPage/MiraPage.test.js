import React from "react";
import { shallow } from "enzyme";
import axios from "axios";
import MiraPage from "./index";
import PrimaryButton from "../../components/PrimaryButton";
import Spinner from "../../components/Spinner";

jest.mock("axios"); // Mock axios module

describe("MiraPage component", () => {
  const projectID = "mockProjectID";

  beforeEach(() => {
    axios.post.mockResolvedValue(); // Mock the axios post method
  });

  it("should render the MiraPage component", () => {
    const wrapper = shallow(<MiraPage projectID={projectID} />);
    expect(wrapper.exists()).toBe(true);
  });

  it("should trigger deployment on button click", () => {
    const wrapper = shallow(<MiraPage projectID={projectID} />);
    const button = wrapper.find(PrimaryButton);

    // Simulate button click
    button.simulate("click");

    // Check if axios.post is called with the correct arguments
    expect(axios.post).toHaveBeenCalledWith(
      expect.stringContaining("/containerize"),
      expect.any(FormData)
    );
  });

  it("should display spinner while loading", () => {
    const wrapper = shallow(<MiraPage projectID={projectID} />);
    wrapper.find(PrimaryButton).simulate("click");
    wrapper.update();

    // Check if the Spinner component is rendered
    expect(wrapper.find(Spinner).exists()).toBe(true);
  });
  it("should match the snapshot", () => {
    const wrapper = shallow(<MiraPage projectID={projectID} />);
    expect(wrapper).toMatchSnapshot();
  });
});
