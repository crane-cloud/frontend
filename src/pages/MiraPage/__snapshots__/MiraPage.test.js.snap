// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiraPage component should match the snapshot 1`] = `ShallowWrapper {}`;
